repository/gorm/message.go

```diff
@@ -294,35 +296,114 @@ func (repo *Repository) SetMessageUnread(userID, messageID uuid.UUID, noticeable
 	if userID == uuid.Nil || messageID == uuid.Nil {
 		return repository.ErrNilID
 	}
+	s := set.UUID{}
+	if noticeable {
+		s.Add(userID)
+	}
+	return repo.BulkSetMessageUnread([]uuid.UUID{userID}, messageID, s)
+}
+
+// BulkSetMessageUnread implements MessageRepository interface.
+func (repo *Repository) BulkSetMessageUnread(userIDs []uuid.UUID, messageID uuid.UUID, noticeable set.UUID) error {
+	if userIDs == nil || messageID == uuid.Nil {
+		return repository.ErrNilID
+	}
+	if len(userIDs) == 0 {
+		return nil
+	}
+	for _, id := range userIDs {
+		if id == uuid.Nil {
+			return repository.ErrNilID
+		}
+	}
```

Comment:
`slices.Contains`でよりシンプルに書けます
```suggestion
	if slices.Contains(userIDs, uuid.Nil) {
		return repository.ErrNilID
	}
```